test/fork/ConstantProductHelper.t..sol

```diff
@@ -0,0 +1,183 @@
+
+    IERC20 private USDC = IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    IERC20 private WETH = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    address private wethUsdcAmm = 0x301076c36E034948A747BB61bAB9CD03f62672e3;
+
+    ConstantProductHelper helper;
+
+    function setUp() public {
+        helper = new ConstantProductHelper();
+        vaultRelayer = address(settlement.vaultRelayer());
+    }
+
+    function testOrder() public {
+        uint256 ammWethInitialBalance = WETH.balanceOf(address(wethUsdcAmm));
+        uint256 ammUsdcInitialBalance = USDC.balanceOf(address(wethUsdcAmm));
+
+        // We send funds to the settlement contract to be able to settle the AMM
+        // rebalancing order without counterpart.
+        uint256 wethBuffer = 1_000_000 ether;
+        vm.deal(address(settlement), wethBuffer);
+        // Wrap ETH
+        vm.prank(address(settlement));
+        (bool result,) = address(WETH).call{value: wethBuffer}(hex"");
+        require(result, "ETH wrapping failed");
+        assertGe(WETH.balanceOf(address(settlement)), wethBuffer);
+
+        // Verify that the tokens are the expected ones.
+        IERC20[] memory tokens = addressVecToIerc20Vec(helper.tokens(wethUsdcAmm));
+        uint256 usdcIndex = 0;
+        uint256 wethIndex = 1;
+        assertEq(tokens.length, 2);
+        assertEq(address(tokens[usdcIndex]), address(USDC));
+        assertEq(address(tokens[wethIndex]), address(WETH));
+
+        // Prepare the price vector used in the execution of the settlement in
+        // CoW Protocol. We skew the price by ~5% towards a cheaper WETH, so
+        // that the AMM wants to buy WETH.
+        uint256[] memory prices = new uint256[](2);
+        // Note: oracle price are expressed in the same format as prices in
+        // a call to `settle`, where the  price vector is expressed so that
+        // if the first token is DAI and the second WETH then a price of 3000
+        // DAI per WETH means a price vector of [1, 3000] (if the decimals are
+        // different, as in WETH/USDC, then the atom amount is what counts).
+        prices[usdcIndex] = WETH.balanceOf(wethUsdcAmm);
+        prices[wethIndex] = USDC.balanceOf(wethUsdcAmm) * 95 / 100;
+
+        // Prepare the vector storing the one trade to be used in the
+        // settlement.
+        GPv2Trade.Data[] memory trades = new GPv2Trade.Data[](1);
+
+        // The helper generates the AMM order
+        GPv2Order.Data memory ammOrder;
+        GPv2Interaction.Data[] memory preInteractions;
+        GPv2Interaction.Data[] memory postInteractions;
+        bytes memory sig;
+        (ammOrder, preInteractions, postInteractions, sig) = helper.order(wethUsdcAmm, prices);
+        // The signature is valid for the contract but not for the
+        // settlement contract. We need to prepend the verifying contract
+        // address.
+        // DISCUSS: should we change that?
+        sig = abi.encodePacked(wethUsdcAmm, sig);
```

Comment:
This required some debugging before I understood why the test was failing. Why doesn't the helper interface return a signature that's ready to use by the settlement contract but it returns some intermediate signature?

Reply:
> Why doesn't the helper interface return a signature that's ready to use by the settlement contract but it returns some intermediate signature?

Because it's an oversight :sweat_smile: Agreed, yes we should change this.

Reply:
Updated the comment. Will address this in a different PR.

Reply:
This is done in #95.